### PR TITLE
Downlaod ticked geometries as one single KML (GSNGM-463)

### PR DESCRIPTION
**Pull request by vladyslav-tk**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -122,3 +122,17 @@ export function filterCsvString(str, bbox4326) {
   const filteredString = lines.join('\n');
   return filteredString;
 }
+
+/**
+ * Check is string can be parsed to object
+ * @param string
```
Comment (fredj) [inline src/utils.js:128]:
```suggestion
 * @param {string} string
```

File: src/cesiumutils.js [hunk 2]
```diff
@@ -218,3 +218,26 @@ export function cartesianToDegrees(cartesian) {
...
 }
+
+/**
+ * Extend kml for export with entities properties
+ * @param {string} kml - kml for export
+ * @param {Array<Entity>} entities - list of entities for export
+ * @return {string}
+ */
+export function extendKmlWithProperties(kml, entities) {
+  entities.values.forEach(ent => {
```
Comment (gberaudo) [inline src/cesiumutils.js:229]:
```suggestion
  entities.values.forEach(entity => {
```

File: src/cesiumutils.js [hunk 3]
```diff
@@ -218,3 +218,26 @@ export function cartesianToDegrees(cartesian) {
...
+    let kmlProperties = '<ExtendedData>';
+    ent.properties.propertyNames.forEach(prop => {
+      let value = ent.properties[prop] ? ent.properties[prop].getValue() : undefined;
+      if (value || typeof value === 'number') {
+        value = typeof value === 'object' ? JSON.stringify(value) : value;
```
Comment (gberaudo) [inline src/cesiumutils.js:234]: I think this is a bit convoluted.
What about something like?
```js
 const valueType = typeof value;
if (valueType === 'number') {
  kmlProperties += `<Data name="${prop}"><value>${value}</value></Data>`;
} else if (valueType === 'object') {
  kmlProperties += `<Data name="${prop}"><value>${JSON.stringify(value)}</value></Data>`;
}
```

And what about string values or arrays?

File: src/utils.js [hunk 4]
```diff
@@ -122,3 +122,17 @@ export function filterCsvString(str, bbox4326) {
...
+ * @returns {boolean}
+ */
+export function isStringJson(string) {
```
Comment (gberaudo) [inline src/utils.js:131]: -> `isSerializedJson`?

File: src/cesiumutils.js (hunk 3 above)
Comment (vladyslav-tk) [reply src/cesiumutils.js]: we don't need to check for 'number', it was only for '0' value. Arrays have type 'object'.

I update the condition to make it more clear

File: src/utils.js (hunk 4 above)
Comment (vladyslav-tk) [reply src/utils.js]: I updated this function. Now, it returns parsed json or undefined to skip the second parse

**Review by gberaudo — APPROVED** [review]

**Pull request merged**